File: Java/Easy/0009-palindrome-number.java
class PalindromeNumber {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int y = x % 10;
        int temp = x / 10;
        while (temp > 0) {
            y *= 10;
            y += (temp % 10);
            temp /= 10;
        }
        return x == y;
    }
}

File: JavaScript/Easy/0009-palindrome-number.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    if (x < 0) {
        return false;
    }
    let y = x % 10;
    let temp = Math.floor(x / 10);
    while (temp > 0) {
        y *= 10;
        y += temp % 10;
        temp = Math.floor(temp / 10);
    }
    return x == y;
};